backend/routes/contactCreator.ts

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import { type User } from '@prisma/client'
+import express from 'express'
+import prisma from '../utils/prismaClient'
+
+const contactCreatorRouter = express.Router()
+const db = prisma
+
+contactCreatorRouter.get('/', async function (req, res) {
+  try {
+    const contacts = await db.contact.findMany({
+      where: { ownerID: (req.user as User).id },
+      include: {
+        relationships: true,
+        importantDateEvent: true
+      }
+    })
+    res.status(200).send({ contacts })
+  } catch (error) {
+    console.error(error)
+    res.status(500).send({ message: 'Failed to retrieve contacts' })
+  }
+})
+
+contactCreatorRouter.post('/', async function (req, res) {
+  try {
+    const contact = await db.contact.create({
+      data: {
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        email: req.body.email,
+        phoneNumber: req.body.phoneNumber,
+        notes: req.body.notes,
+        ownerID: (req.user as User).id,
+        source: req.body.source
+      }
+    })
+    await db.userContactRelationship.createMany({
+      data: req.body.relationships.map((relationship: { relationshipType: any }) => ({
```

Comment:
let's try and provide more useful types than `any`

Reply:
sounds good!